modules/lib/api/document.xql

```diff
@@ -326,7 +326,8 @@ declare function dapi:epub($request as map(*)) {
 };
 
 declare %private function dapi:work2epub($request as map(*), $id as xs:string, $work as document-node(), $lang as xs:string?) {
-    let $config := $config:epub-config($work, $lang)
+    let $navRoot := fn:fold-left(tokenize($request?parameters?nav-root, '/'), (), function($a, $b) { (if (not(empty($a))) then $a else $work)/node()[position()=xs:integer($b)] })
```

Comment:
The TEI Publisher application makes extensive use of the functions `util:node-id()` and `util:node-by-id()`. I suspect they may exhibit better performance. Were you aware of these functions? If so, did you have a reason for opting not to use them?

Reply:
I chose to use the position of the node instead because the node to be the root for navigation will most likely be `text` or `body`, which don't always have an ID

Reply:
The `util:node-id()` and `util:node-by-id()` functions do use the node's numeric position in the node hierarchy. For their function documentation, see https://exist-db.org/exist/apps/fundocs/view.html?uri=http://exist-db.org/xquery/util#node-by-id.2. 

For example, given the sample TEI file from your test saved to `/db/test/test.xml`, the query `util:node-id(doc("/db/test/test.xml")/tei:TEI/tei:text)` returns `1.4`, and correspondingly, the query `util:node-by-id(doc("/db/test/test.xml"), "1.4")` returns the `<text>` element.

Reply:
I haven't thought of using that, I will test it and let you know.